src/API.jl

```diff
@@ -989,7 +989,9 @@ function precompile(ctx::Context; internal_call::Bool=false, kwargs...)
     num_tasks = parse(Int, get(ENV, "JULIA_NUM_PRECOMPILE_TASKS", string(Sys.CPU_THREADS::Int + 1)))
     parallel_limiter = Base.Semaphore(num_tasks)
     io = ctx.io
+
     fancyprint = can_fancyprint(io)
+    @debug fancyprint = false # disable fancy printing in debug mode
```

Comment:
I don't think this works, there is no guarantee that this code will be executed. TBH it feels like a bug that you can introduce local vars like this.